documentation/wallet.adoc

```diff
@@ -0,0 +1,853 @@
+|===
+
+.Data Usage (Beyond Circulation)
+
+Apart from the regular data flow, certain data elements are used in specific scenarios or by specific components:
+
+[%header,cols="1,3,1"]
+|===
+| Element | Description | Data Involved
+| Core | Processes various requests. | Private Key, Password
+|===
+
+== Functional Requirements
+
+=== API
+
+The API will adhere to the HTML REST JSON standard, utilizing classic HTTP responses to indicate the status and nature of each response.
+
+NOTE: For hands-on testing and examples, refer to the provided Swagger documentation and consider using tools like Postman.
+
+==== Sign Operation
+
+The 'sign' endpoint serves the purpose of signing all five operation 
+types, including Transaction, Roll buy/sell, CallSC, and 
+ExecuteSC. It provides the capability to batch operation requests by 
+utilizing the correlationID parameter. No user interaction is required once first operation processing is underway. 
+Batching operation requests in Massa Wallet does not entail processing the operations as a group. 
+Instead, it enhances the user experience by allowing users to, sequentially, sign multiple operations 
+after being prompted to provide a password once.
+
+===== Parameters
+
+[%header,cols="1,3,1,1"]
+|===
+| Name | Description | Presence | Format
+| `nickname` | Identifier for the account. | Required | String
+| `description` | Text describing the signing intent. | Optional | String (Default: "", Max: 280 characters)
+| `operation` | Serialized attributes for signing. | Required | Byte string
+| `batch` | Specifies if the operation is part of a batch. | Optional | Boolean (Default: `false`)
+| `correlationId` | Identifier for correlating multiple requests. | Optional | String (Default: "")
+|===
+
+
+IMPORTANT: To ensure security, the `correlationId` should be sent in the request body, not in the query string.
+
+===== Response
+
+On successful execution, the response will include an array containing: signature, public key, and the correlationId will be included if the batch parameter is set to true.
+
+==== Sign Message
+
+This endpoint facilitates the signing of a message using a 
+specific account resource, identified by its unique nickname. It's 
+important to note that this endpoint is not idempotent: identical 
+calls with the same data will initiate separate executions.
+
+===== Parameters
+
+[%header,cols="1,3,1,1"]
+|===
+| Name | Description | Presence | Format
+| `nickname` | Identifier for the account. | Required | String
+| `description` | Text describing the signing intent. | Optional | String (Default: "", Max: 280 characters)
+| `message` | **tbd** | Optional/Required |**define type**
+| `displayData` | **tbd** | **tbd** | **tbd**
+|===
+
+
+===== Response
+
+On successful execution, the response will return the signature and public key of the signed message.
+
+=== Functions
```

Comment:
Should be aligned with what was done for the lazy manager. Furthermore, a lot of seuquence diagrams were added, but without the corresponding functions. Both sections should growth in parallel.